src/libraries/Fuzzing/DotnetFuzzing/PooledBoundedMemory.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+
+namespace DotnetFuzzing;
+
+internal sealed class PooledBoundedMemory<T> : IDisposable where T : unmanaged
+{
+    // Default libFuzzer max_len for inputs is 4096.
+    private const int MaxLength = 4096 * 2;
```

Comment:
Isn't pooling in this fashion potentially going to allocate dozens of GB of memory?